spec/common/modules/testapp_manager.js

```diff
@@ -1,15 +1,18 @@
 "use strict";
+/* global define, isNativescript, fetch */
 
 /* testapp module is responsible for setting up and tearing down apps in the test environment */
 define(['globals', 'browser-base64', 'ably'], function(ablyGlobals, base64, ably) {
 	var restHost = ablyGlobals.restHost || prefixDomainWithEnvironment('rest.ably.io', ablyGlobals.environment),
 			tlsPort  = ablyGlobals.tlsPort;
 
+
 	var isBrowser = (typeof(window) === 'object'),
 			httpReq   = httpReqFunction(),
-			toBase64  = base64Function(),
-			loadJsonData = isBrowser ? loadJsonDataBrowser : loadJsonDataNode,
-			testResourcesPath = (isBrowser && window.__karma__ && window.__karma__.start ? 'base/' : '') + 'spec/common/ably-common/test-resources/';
+			toBase64  =  base64Function(),
+			loadJsonData = isNativescript ? loadJsonNativescript : isBrowser ? loadJsonDataBrowser : loadJsonDataNode,
+			testResourcesPath = isNativescript ? '~/ably/ably-common/test-resources/' :
```

Comment:
Sorry, I think you misunderstand where this will be running...  :-)

  `~` = the devices running application home path.  i'e.  If the app is named `com.ably.demo`  The files will be located at `/data/data/com.ably.demo/files/app/...` on an android device and on an iOS device the path is also a big old deep path that I don't really want to type  ;-) .     That is NOT the home path on linux/mac.     This is the correct, and it is a fixed path, and is ONLY for testing inside a NativeScript app.    However your second point; is accurate; now that I understand the layout of how your testing system works; I can actually direct the test data to the node_modules/ably folder without an issue.   Good catch, "my" layout has been in place since I first cloned all your tests into the app.

-  Your second point, part of it is correct some of these files are cloned due to the limitation for how NativeScript test framework works, and also the way you use nodeunit&define.   But we can do away with the testing data; those can be directed to the originals and not copied, along with pretty much everything else...   

3. You are correct, I can't believe I did that, I know better -- I'll update that code.  Thanks.

4. I hate them also; I just was quickly trying to fix it, months ago. :)  But your right technical debt is not a good thing.  I'll clean that area up...